ocaml/typing/datarepr.ml

```diff
@@ -111,6 +111,17 @@ let constructor_descrs ~current_unit ty_path decl cstrs rep =
     | Variant_extensible -> assert false
     | Variant_boxed x -> x
     | Variant_unboxed -> [| Constructor_uniform_value, [| decl.type_jkind |] |]
+    | Variant_with_null ->
+      (* CR layouts v3.5: this hardcodes ['a or_null]. Fix when we allow
+         users to write their own null constructors. *)
+      (* CR layouts v3.3: generalize to [any_non_null]. *)
```

Comment:
What's the generalization you imagine?

Reply:
`('a : any_non_null) or_null`, with the jkind of `or_null` matching the jkind of `'a` but `maybe_null`. We can't express this in the type system yet.

Reply:
Also, this maybe should be v3.4, not v3.3.